reports/generator.go

```diff
@@ -140,11 +151,17 @@ func (r *report) htmlToPDF(ctx context.Context, htmlContent string) ([]byte, err
 	return pdfBytes, nil
 }
 
-func formatTime(t float64) string {
+func formatTimeWithTimezone(t float64, timezone string) string {
+	loc := resolveTimezone(timezone)
+
+	var timeVal time.Time
 	if t > 9999999999 {
-		return time.Unix(0, int64(t)).Format("2006-01-02 15:04:05")
+		timeVal = time.Unix(0, int64(t)).In(loc)
+	} else {
+		timeVal = time.Unix(int64(t), 0).In(loc)
 	}
-	return time.Unix(int64(t), 0).Format("2006-01-02 15:04:05")
+
+	return timeVal.Format("2006-01-02 15:04:05")
```

Comment:
Avoid using `else` for the sake of consistency, avoid magic numbers; there is probably a constant (`>= time.Millisecond` or something).